packages/multichain/src/caip25Permission.ts

```diff
@@ -0,0 +1,283 @@
+import type { NetworkClientId } from '@metamask/network-controller';
+import type {
+  PermissionSpecificationBuilder,
+  EndowmentGetterParams,
+  ValidPermissionSpecification,
+  PermissionValidatorConstraint,
+  PermissionConstraint,
+} from '@metamask/permission-controller';
+import {
+  CaveatMutatorOperation,
+  PermissionType,
+} from '@metamask/permission-controller';
+import type { CaipAccountId, Json } from '@metamask/utils';
+import {
+  parseCaipAccountId,
+  type Hex,
+  type NonEmptyArray,
+} from '@metamask/utils';
+import { strict as assert } from 'assert';
```

Comment:
Note that this makes this module dependent on Node (or at least, it means that whatever bundler the consumer is using needs to fill in this module). Is that intentional? Is there a way to use plain JavaScript instead?

Reply:
not intentional. thanks for pointing this out. Replaced with lodash.isEqual here https://github.com/MetaMask/core/pull/4784/commits/d75d5d69652177e2942c7542f8e7d10194782bca